packages/access-api/src/models/spaces.js

```diff
@@ -0,0 +1,97 @@
+// @ts-ignore
+// eslint-disable-next-line no-unused-vars
+import * as Ucanto from '@ucanto/interface'
+import { delegationToString } from '@web3-storage/access/encoding'
+import { Kysely } from 'kysely'
+import { D1Dialect } from 'kysely-d1'
+import { D1Error, SpacePlugin } from '../utils/d1.js'
+
+const spacePlugin = new SpacePlugin()
+
+/**
+ * Spaces
+ */
+export class Spaces {
+  /**
+   *
+   * @param {D1Database} d1
+   */
+  constructor(d1) {
+    this.d1 = /** @type {Kysely<import('../bindings').D1Schema>} */ (
+      new Kysely({ dialect: new D1Dialect({ database: d1 }) })
+    )
+  }
+
+  /**
+   * @param {import('@web3-storage/capabilities/types').VoucherRedeem} capability
+   * @param {Ucanto.Invocation<import('@web3-storage/capabilities/types').VoucherRedeem>} invocation
+   * @param {Ucanto.Delegation<[import('@web3-storage/access/types').Top]> | undefined} delegation
+   */
+  async create(capability, invocation, delegation) {
+    try {
+      const metadata =
+        /** @type {import('@web3-storage/access/types').SpaceTableMetadata | undefined} */ (
+          /** @type {unknown} */ (invocation.facts[0])
+        )
+      const result = await this.d1
+        .withPlugin(spacePlugin)
+        .insertInto('spaces')
+        .values({
+          agent: invocation.issuer.did(),
+          did: capability.nb.space,
+          email: capability.nb.identity.replace('mailto:', ''),
+          invocation: await delegationToString(invocation),
+          product: capability.nb.product,
+          metadata,
+          delegation: !delegation
+            ? undefined
+            : await delegationToString(delegation),
+        })
+        .returning('spaces.did')
+        .execute()
+      return { data: result }
+    } catch (error) {
+      return {
```

Comment:
Why return error and not throw?

Reply:
cause i need to use the error on the provider side and i dont want to try catch the full provider logic. im working on normalising this all thing in the w3space/account refactor